src/Compilers/Core/Portable/Text/SourceText.cs

```diff
@@ -724,25 +724,46 @@ public virtual string ToString(TextSpan span)
             CheckSubSpan(span);
 
             // default implementation constructs text using CopyTo
-            var builder = PooledStringBuilder.GetInstance();
-            var buffer = s_charArrayPool.Allocate();
+            var tempBuffer = s_charArrayPool.Allocate();
+            string result;
 
             int position = Math.Max(Math.Min(span.Start, this.Length), 0);
             int length = Math.Min(span.End, this.Length) - position;
+
+#if NET
+            result = string.Create(length, (this, position, length, tempBuffer), static (buffer, arg) =>
+            {
+                var (sourceText, position, length, tempBuffer) = arg;
+
+                while (buffer.Length > 0)
+                {
+                    int copyLength = Math.Min(tempBuffer.Length, length);
+                    sourceText.CopyTo(position, tempBuffer, 0, copyLength);
```

Comment:
Sorry, I'm not sure what I'm missing here; what ensures that this is safe? What if `copyLength` is longer than `tempBuffer`?

Reply:
Was this comment made against commit 1 or 2?

Reply:
Commit 1. Glad I wasn't going crazy :laughing:.